ui/apps/ui/src/app/components/download-results/download-results-button.component.ts

```diff
@@ -0,0 +1,130 @@
+import { Component, ViewChild } from '@angular/core';
+import { UntilDestroy } from '@ngneat/until-destroy';
+import { DownloadResultsModalComponent } from '@components/download-results/download-results-modal.component';
+import { filter, first, switchMap } from 'rxjs';
+import {
+  ICollectionSearchMetadata,
+  IDownloadResult,
+  downloadAdapterType,
+} from '@collections/repositories/types';
+import { queryChanger } from '@collections/filters-serializers/utils';
+import { Parser } from '@json2csv/plainjs';
+import { saveAs } from 'file-saver-es';
+import { CustomRoute } from '@collections/services/custom-route.service';
+import { ActivatedRoute, Router } from '@angular/router';
+import { FetchDataService } from '@collections/services/fetch-data.service';
+import { SearchMetadataRepository } from '@collections/repositories/search-metadata.repository';
+import { AdaptersRepository } from '@collections/repositories/adapters.repository';
+
+const DEFAULT_NO_OF_RESULTS = 2000;
+
+@UntilDestroy()
+@Component({
+  selector: 'ess-download-results-button',
+  template: `
+    <div
+      class="download-button"
+      (click)="downloadResults()"
+      [ngbTooltip]="'Download first 2000 results as a CSV file'"
+    >
+      <img
+        class="download-icon"
+        src="assets/download-button.svg"
+        alt="download-icon"
+      />
+      Download Results
+    </div>
+
+    <ess-download-results-modal
+      [loading]="modalLoading"
+      [error]="modalError"
+    ></ess-download-results-modal>
```

Comment:
I'm not the biggest fan of how modals were handled here.
It would be better to have modal component which would be passed directly to `this.modalRef = this.modalService.open(DownloadResultsModalComponent)`, and you could still manipulate inputs via `this.modalRef. componentInstance.loading = false`.

But it would be more readable. Creating component which does not have real html template, only `ng-template` which is being passed to the modal is a little bit working around.